_posts/2022-10-01-arrow-parquet-encoding-part-2.md

```diff
@@ -0,0 +1,344 @@
+ │  └─────┘               │    │  │  1  │             │ │  │  6  │  │ │
+ │                        │  │ │  └─────┘             │ │  └─────┘  │
+ │  Definition    Data    │    │                      │ │           │ │
+ │    Levels              │  │ │  Definition   Data   │ │   Data    │
+ │                        │    │    Levels            │ │           │ │
+ │  "a"                   │  │ │                      │ │           │
+ └────────────────────────┘    │  "b.b1"              │ │  "b.b2"   │ │
+                             │ └──────────────────────┘ └───────────┘
+                                  "b"                                 │
+                             └ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─
+
+
+┌ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─  ┌ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─
+  ┌──────────────────────┐ │   ┌──────────────────────┐ ┌──────────────────────┐  │
+│ │  ┌─────┐    ┌─────┐  │   │ │  ┌─────┐    ┌─────┐  │ │ ┌─────┐     ┌─────┐  │
+  │  │  0  │    │  6  │  │ │   │  │  1  │    │  1  │  │ │ │  1  │     │  1  │  │  │
+│ │  ├─────┤    ├─────┤  │   │ │  ├─────┤    ├─────┤  │ │ ├─────┤     └─────┘  │
+  │  │  1  │    │  7  │  │ │   │  │  1  │    │  2  │  │ │ │  2  │              │  │
+│ │  ├─────┤    └─────┘  │   │ │  ├─────┤    └─────┘  │ │ ├─────┤              │
+  │  │  1  │             │ │   │  │  0  │             │ │ │  0  │              │  │
+│ │  └─────┘             │   │ │  └─────┘             │ │ └─────┘              │
+  │                      │ │   │                      │ │                      │  │
+│ │  Definition   Data   │   │ │  Definition   Data   │ │ Definition   Data    │
+  │    Levels            │ │   │    Levels            │ │   Levels             │  │
+│ │                      │   │ │                      │ │                      │
+  │  "c.1"               │ │   │  "d.1"               │ │  "d.d2"              │  │
+│ └──────────────────────┘   │ └──────────────────────┘ └──────────────────────┘
+     "c"                   │      "d"                                             │
+└ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─  └ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─ ─
+ ```
+
+## List / Repeated Columns
+
+Closing out support for nested types are *lists*, which containing a variable number of other values. For example,
+
+```json
+{                     <-- First record
+  "a": [1],           <-- top-level field a containing list of integers
+}
+```
+```json
+{                     <-- "a" is not provided (is null)
+}
+```
+```json
+{                     <-- "a" is non-null but empty
+  "a": []
+}
+```
+```json
+{
+  "a": [null, 2],     <-- list elements can themselves be null
+}
+```
+
+Documents of this format could be stored in this Arrow schema
+
+```text
+Field(name: "a", nullable: true, datatype: List(
+  Field(name: "element", nullable: true, datatype: Int32),
+)
+```
+
+Documents of this format could be stored in this Parquet schema
```

Comment:
I would move this down to the parquet section, same as was done for the section above